src/extension/feature-pregeneralized/src/main/java/org/geoserver/data/gen/info/GeneralizationInfosProviderImpl.java

```diff
@@ -50,4 +58,39 @@ protected URL deriveURLFromSourceObject(Object source) throws IOException {
         }
         throw new IOException("Cannot read from " + source);
     }
+
+    @Override
+    protected GeneralizationInfos parseXML(URL url) throws IOException {
+
+        GeoServerResourceLoader loader = GeoServerExtensions.bean(GeoServerResourceLoader.class);
+        Resource configurationResource = loader.fromURL(url);
+
+        Document doc = null;
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setIgnoringComments(true);
+        factory.setNamespaceAware(true);
+        factory.setIgnoringElementContentWhitespace(true);
+
+        try {
+            DocumentBuilder db = factory.newDocumentBuilder();
+            doc = db.parse(configurationResource.in());
```

Comment:
Rather declare the `InputStream` in the enclosing try to make it a try-with-resources block (`try(InputStream in = configurationResource.in()){...` so it gets closed

Reply:
Done